main/fastcgi.c

```diff
@@ -1469,6 +1474,7 @@ int fcgi_accept_request(fcgi_request *req)
 			return -1;
 		}
 		if (fcgi_read_request(req)) {
+			req->hook.on_read();
```

Comment:
I'm probably horribly misunderstanding something here... But isn't the connection also already there is in the case where initially fd was <0? After all, the poll/select is run *after* the accept(), not before it.